plugwise/helper.py

```diff
@@ -239,11 +243,98 @@ def all_locations(self):
         return
 
+    def all_appliances_class_finder(self, appliance, appl):
+        """Determine class per appliance."""
+        # Find gateway and heater_central devices
+        if appl.pwclass == "gateway":
+            self.gateway_id = appliance.attrib["id"]
+            appl.fw = self.smile_version[0]
+            appl.model = appl.name = self.smile_name
+            appl.v_name = "Plugwise B.V."
+            return appl
+
+        if appl.pwclass in THERMOSTAT_CLASSES:
+            locator = ".//logs/point_log[type='thermostat']/thermostat"
+            mod_type = "thermostat"
+            module_data = self.get_module_data(appliance, locator, mod_type)
+            appl.v_name = module_data[0]
+            appl.model = check_model(module_data[1], appl.v_name)
+            appl.fw = module_data[3]
+            return appl
+
+        if appl.pwclass == "heater_central":
+            # Remove heater_central when no active device present
+            if not self.active_device_present:
+                return None
+
+            self.heater_id = appliance.attrib["id"]
+            appl.name = "Auxiliary"
+            locator1 = ".//logs/point_log[type='flame_state']/boiler_state"
+            locator2 = ".//services/boiler_state"
+            mod_type = "boiler_state"
+            module_data = self.get_module_data(appliance, locator1, mod_type)
+            if module_data == [None, None, None, None]:
+                module_data = self.get_module_data(appliance, locator2, mod_type)
+            appl.v_name = module_data[0]
+            appl.model = check_model(module_data[1], appl.v_name)
+            if appl.model is None:
+                appl.model = (
+                    "Generic heater/cooler" if self._cp_state else "Generic heater"
+                )
+            return appl
+
+        if self.stretch_v2 or self.stretch_v3:
+            locator = ".//services/electricity_point_meter"
+            mod_type = "electricity_point_meter"
+            module_data = self.get_module_data(appliance, locator, mod_type)
+            appl.v_name = module_data[0]
+            if appl.model != "Group Switch":
+                appl.model = None
+            if module_data[2] is not None:
+                hw_version = module_data[2].replace("-", "")
+                appl.model = version_to_model(hw_version)
+            appl.fw = module_data[3]
+            return appl
+
+        if self.smile_type != "stretch" and "plug" in appl.types:
+            locator = ".//logs/point_log/electricity_point_meter"
+            mod_type = "electricity_point_meter"
+            module_data = self.get_module_data(appliance, locator, mod_type)
+            appl.v_name = module_data[0]
+            appl.model = version_to_model(module_data[1])
+            appl.fw = module_data[3]
+            return appl
+
+        # Cornercase just return existing dict-object
+        return appl  # pragma: no cover
+
+    def all_appliances_types_finder(self, appliance, appl):
```

Comment:
I would suggest to remove `all_` from the function name.

Reply:
see above, the reasoning being them partials - but i'm all up for shortening the names